### ISSUE TITLE
Prepare data for NGWOS in/out experiment

### ISSUE THREAD

**Issue by aappling-usgs** [USGS-R/delaware-model-prep#28]
The goal of this experiment is to get a first look at the utility of the NGWOS data in improving water temperature predictions in the DRB. There need only be two model training+prediction runs.

For both runs:
* hold out all data from the Lordsville gage and maybe one or two other long-term monitoring sites
* hold out data from the first ~third of the dataset for all sites
* train in the same way for both runs - for example, pretrain on both temperature and discharge from SNTemp, finetune on both temperature and discharge observations.
* generate test predictions for the held-out data. In the very near term, I'm most interested in the comparison of accuracy at Lordsville in ~2017-2019 for each of the two model runs, and secondarily in Lordsville accuracy in the test time period (the first ~third of the dates) and in the pre-NGWOS period (e.g., 2000-2015 or so)

With-NGWOS treatment:
* for training, use all available data (including data from NGWOS sites) except the held-out data noted above

Without-NGWOS treatment:
* exclude all NGWOS data, i.e., data from NGWOS sites and present in 2017+. keep data from "NGWOS" sites from before 2017. exclude the held-out data noted above.

Later we could expand to look at discharge (might be a freebee with whatever has already been run), consider the location and number of held out long-term sites, consider prediction accuracy during 2017-20 vs earlier years, etc. But the above first-cut experiment with just two model runs should be plenty to get us thinking and talking usefully about the role of NGWOS data in our modeling.

**Comment by limnoliver** [comment]
What does first third mean? Setting test period to 1980-1995 or so?

**Comment by aappling-usgs** [comment]
Yes, I was thinking 1980-1995 but couldn't remember exactly when the input data started and ended. The key is just to have a test period that doesn't exclude all the NGWOS data from both sites (i.e., 2017-2019 can't be in the test period)

**Comment by aappling-usgs** [comment]
Which raises the important point that Jeff needs to be using Jake's most recent SNTemp outputs / PGDL inputs so we can include 2019 data in the training.

**Comment by jsadler2** [comment]
Right. I have Jakes new SNTemp io through 2019

**Comment by jsadler2** [comment]
For "excluding NGWOS" data, we are just excluding the reaches where NGWOS stations were so (from Sam):
```R
c("1556", "1570", "1569", "1653", "1659", "1485", "1489", 
  "1494", "1762", "1784", "2291", "2338", "1806", "2036", 
  "2034", "2033", "2030", "1437", "1455", "1658", "2303")
```

**Comment by jsadler2** [comment]
And do we have the Lordville reach id handy?

**Comment by jzwart** [comment]
Lordville `seg_id_nat` is 1573

**Comment by aappling-usgs** [comment]
@limnoliver has picked out three ~long-term gages including Lordville to hold out - IDs of the other two forthcoming, I assume.

**Comment by limnoliver** [comment]
Also putting this here for our records and for @jsadler2:

NGWOS seg_id_nat = `c("1556", "1570", "1569", "1653", "1659", "1485", "1489", "1494", "1762", "1784", "2291", "2338", "1806", "2036", "2034", "2033", "2030", "1437", "1455", "1658", "2303")`

Lordville is seg_id_nat 1573.

seg_id_nat to withhold = `c("1573",  "1568", "1718")

**Comment by aappling-usgs** [comment]
> For "excluding NGWOS" data, we are just excluding the reaches where NGWOS stations were

Yes but only exclude data from Oct 2017 and later for those reaches

**Comment by jzwart** [comment]
Do we have a map of NGWOS sites already? I'm interested in how close those are to the withheld sites

**Comment by jsadler2** [comment]
> Yes but only exclude data from Oct 2017 and later for those reaches

Okay. I'll have to adjust my code. Currently it's set up to disregard all or nothing from a station.

**Comment by aappling-usgs** [comment]
Map of NGWOS sites from Sam:
![MicrosoftTeams-image](https://user-images.githubusercontent.com/12039957/79239217-4d68b000-7e3e-11ea-955b-7d73d5ede027.png)

**Comment by limnoliver** [comment]
Map here: yellow highlighted are the withheld sites

![image](https://user-images.githubusercontent.com/15788176/79239250-f6f77380-7e35-11ea-95bc-76a0ee22e932.png)

**Comment by limnoliver** [comment]
Temperature observations from NGWOS sites - vertical red line is Oct 1, 2017

![image](https://user-images.githubusercontent.com/15788176/79240453-6d48a580-7e37-11ea-9cea-78210288887d.png)

**Comment by limnoliver** [comment]
@jsadler2  - SNTEMP predictions begin on 1980-10-01 and end on 2019-12-31. A third of that time period is ~4778 days. With some rounding, a close first third would be 1980-10-01 to 1993-10-01 (actual third would end on 1993-10-31, but I like the round 13 years of data)

**Comment by aappling-usgs** [comment]
@jsadler2 From @jzwart : also exclude the oddball SNTemp prediction (the only one greater than 40 C, and it's much greater) on 2018-09-30 for seg_id_nat 2045

**Comment by limnoliver** [comment]
Oops - maybe we both posted and then deleted our maps? Again here for reference: 
![MicrosoftTeams-image](https://user-images.githubusercontent.com/15788176/79243042-b8b08300-7e3a-11ea-9b61-22144dbe22ac.png)

**Comment by jsadler2** [comment]
### Results
![MicrosoftTeams-image (2)](https://user-images.githubusercontent.com/6943441/79580186-433cf080-808e-11ea-906d-5c8baea6f50f.png)
![MicrosoftTeams-image (3)](https://user-images.githubusercontent.com/6943441/79580194-46d07780-808e-11ea-912b-4f14463cdfe5.png)
seg 1718
![1718_NGWOS_experiment_timeseries](https://user-images.githubusercontent.com/6943441/79580258-610a5580-808e-11ea-891f-befea137ff90.png)
seg 1568
![1568_NGWOS_experiment_timeseries](https://user-images.githubusercontent.com/6943441/79580288-68c9fa00-808e-11ea-97fb-c08eb12ba69e.png)
seg 1573 (Lordville)
![Lordville_NGWOS_experiment_timeseries](https://user-images.githubusercontent.com/6943441/79580330-77181600-808e-11ea-8c41-177793ec24d2.png)
seg 1806 (one of the NGWOS) (training period)
![1806_NGWOS_experiment_timeseries](https://user-images.githubusercontent.com/6943441/79580387-88612280-808e-11ea-9d39-8282762d7743.png)
seg 1784 (one of the NGWOS) (training period)
![1784_NGWOS_experiment_timeseries](https://user-images.githubusercontent.com/6943441/79580421-9616a800-808e-11ea-936e-8f6a6ef618ca.png)

**Comment by limnoliver** [comment]
@jsadler2 -- the new NGWOS reaches for the updated experiment are (also target `ngwos_reaches` in my latest PR):

`c('1457', '1556', '1575', '1570', '1569', '1634', '1653', '1659', '1667', '1654', '1666', '1485', '1702', '1707', '1701', '1699', '1489', '1494', '1766', '1762', '1790', '1793', '1792', '1781', '1804', '1784', '1815', '2291', '2338', '1806', '2036', '2021', '2034', '2033', '2030', '1437', '1441', '1440', '1449', '1455', '1558', '1644', '1636', '1642', '1637', '1638', '1646', '1648', '2303', '2694', '1765', '1787', '1808', '2021', '2021', '1810')`

**Comment by jsadler2** [comment]
And so these are now included in `obs_temp_drb.rds` on Gdrive?

**Comment by limnoliver** [comment]
Yep - all new data is pushed to GD.

**Comment by jsadler2** [comment]
@limnoliver  - just double checking some things here. I am looking at an `obs_temp_full.csv` file that I copied from `9_collaborator_data/umn/` back in April (modified date, 4/14). All of the stations that you mention above are in this older file. Is that expected?

**Comment by limnoliver** [comment]
Yes, but they probably do not have the most recent NGWOS data.

**Comment by jsadler2** [comment]
Gotcha. 
Spotting checking 1637. 
Old file: 228 rows of data. Max date: 2003-09-09
New file 749 rows of data. Max date: 2020-05-01

**Comment by jsadler2** [comment]
There _are_ 19 stations that are not in the previous one. Is _that_ expected? I suppose those could be the new, more lenient `fish_distance` threshold?
```
array([2696., 2700., 1597., 1599., 1451., 1688., 1690., 1694., 1709.,
       4189., 1768., 1820., 1500., 2278., 2283., 2285., 2300., 2318.,
       2323.])
```

**Comment by limnoliver** [comment]
Oh, and just realized I didn't build this through the cooperator data, so I should do that and push new files/indicators. Will add a small PR.

**Comment by jsadler2** [comment]
I built just that file locally, but I can build the rest too.

**Comment by limnoliver** [comment]
And yes, could be a combo of fish distance AND NGWOS.

Cool, yes, please build and push all the way through.

**Comment by jsadler2** [comment]
Okay. I will build push and make a PR

**Comment by jsadler2** [comment]
> And yes, could be a combo of fish distance AND NGWOS.
But if those were NGWOS, wouldn't they show up in the other list? None of the new ones show up in your "NGWOS" list above...

**Comment by limnoliver** [comment]
oh sorry, yes, correct.

**Comment by jsadler2** [comment]
One last question: is the date threshold the same for withholding the NGWOS data: `dates > Oct. 2017`

**Comment by limnoliver** [comment]
Yes, I think that's fine for now. It's not perfect, but we'll show the data to Brian P and ask if there's a better way to figure out the exact start of what we call "NGWOS".
